### PR TITLE
Make `app.in_foreground` searchable

### PR DESCRIPTION
**Pull request by marandaneto**
Make `app.in_foreground` searchable.

Docs PR: https://github.com/getsentry/develop/pull/728
Relay PR: https://github.com/getsentry/relay/pull/1578
Meta issue: https://github.com/getsentry/team-mobile/issues/36

Follow up https://github.com/getsentry/sentry/pull/44157

### CODE REVIEW

File: src/sentry/snuba/events.py [hunk 1]
```diff
@@ -559,6 +559,23 @@ class Columns(Enum):
...
+    )
+    # TODO: review this
+    THREAD_MAIN = Column(
+        group_name="events.threads.values.main",
+        event_name="threads.values.main",
+        transaction_name="threads.values.main",
+        discover_name="threads.values.main",
+        issue_platform_name="threads.values.main",
+        alias="thread.main",
+    )
```
Comment (marandaneto) [inline src/sentry/snuba/events.py:578]: I don't know if this works, apparently `threads` are not mapped anywhere else, maybe this won't work?

File: src/sentry/rules/conditions/event_attribute.py [hunk 2]
```diff
@@ -184,6 +186,23 @@ def _get_attribute_values(self, event: GroupEvent, attr: str) -> Sequence[str]:
...
+                    response = {}
+                return [response.get(path[1])]
+
+            return []
+        # TODO: review this
+        elif path[0] == "thread":
+            if path[1] not in ("main"):
+                return []
+
+            return [getattr(e, path[1]) for e in event.interfaces["threads"].values]
```
Comment (marandaneto) [inline src/sentry/rules/conditions/event_attribute.py:204]: I don't know if `event.interfaces["threads"]` works, I've used `event.interfaces["exception"]` as an example, `threads` are not being used as of now.

Comment (marandaneto) [reply src/sentry/rules/conditions/event_attribute.py]: Will make another PR

File: src/sentry/snuba/events.py (hunk 1 above)
Comment (marandaneto) [reply src/sentry/snuba/events.py]: Will make another PR

**Review by narsaynorath — APPROVED** [review]
Changes look good to me!

**Pull request merged**